app/src/main/java/org/simple/clinic/di/DateFormatterModule.kt

```diff
@@ -24,8 +24,16 @@ class DateFormatterModule {
 
   @Provides
   @Named("full_date")
-  fun provideDateFormatterForFullDate(locale: Locale, chronology: Chronology): DateTimeFormatter {
-    return DateTimeFormatter.ofPattern("d-MMM-yyyy", locale)
+  fun provideDateFormatterForFullDate(
+      locale: Locale,
+      chronology: Chronology,
+      country: Country
+  ): DateTimeFormatter {
+    val pattern = when (country.isoCountryCode) {
+      Country.ETHIOPIA -> "d-MM-yyyy"
+      else -> "d-MMM-yyyy"
+    }
```

Comment:
Similar feedback as the earlier comment whether we should move this `when` clause to a computed property of the `Country` type. There's a strong case given that this block is being repeated in multiple places.

Reply:
In this case there are different patterns we will be handling. So we would have to move all the patterns (most of them), so the computed properties would like.

```kt
val fullDatePatterns: String
val userInputDatePattern: String
```

Is that fine?

Reply:
Yeah, I think that looks good.